# Bainbridge Island Home Price Prediction
This notebook trains a simple Decision Tree Regressor using home data from Bainbridge Island, WA to predict house prices based on features like bedrooms, bathrooms, square footage, lot size, and year built.

In [20]:
%pip install pandas scikit-learn openpyxl

Note: you may need to restart the kernel to use updated packages.


In [24]:
# Import packages
import pandas as pd
from sklearn.tree import DecisionTreeRegressor
from sklearn.metrics import mean_absolute_error
from sklearn.model_selection import train_test_split

In [4]:
# Homes on Bainbridge Island, WA
home_data = pd.read_excel("./BI_homes.xlsx")

In [5]:
home_data.describe()

bed       bath         sqft   acre_lot   year_built         price
count  45.000000  45.000000    45.000000  45.000000    45.000000  4.500000e+01
mean    3.111111   2.977778  2936.755556   1.505333  1988.777778  2.051282e+06
std     0.982164   1.137958  1842.161788   2.995842    23.356806  1.725451e+06
min     1.000000   1.000000   750.000000   0.030000  1906.000000  4.980000e+05
25%     3.000000   2.000000  1839.000000   0.380000  1975.000000  1.050000e+06
50%     3.000000   3.000000  2358.000000   0.780000  1995.000000  1.498000e+06
75%     4.000000   3.500000  3492.000000   1.130000  2003.000000  2.275000e+06
max     6.000000   5.500000  9377.000000  19.320000  2024.000000  7.950000e+06

In [7]:
y = home_data.price

# Features
X = home_data[["bed", "bath", "sqft", "acre_lot", "year_built"]]

In [25]:
# Split the data into training and validation
train_X, val_X, train_y, val_y = train_test_split(X, y, random_state=1)

In [26]:
# Train the model on training set
home_model = DecisionTreeRegressor(random_state=1)
home_model.fit(train_X, train_y)

DecisionTreeRegressor(random_state=1)

In [39]:
# Predict on validation set
val_predictions = home_model.predict(val_X)
for i, prediction in enumerate(val_predictions, 1):
    print(f"Prediction {i}: ${prediction:,.0f}")

Prediction 1: $1,189,000
Prediction 2: $1,949,000
Prediction 3: $649,000
Prediction 4: $7,296,000
Prediction 5: $3,998,000
Prediction 6: $1,875,000
Prediction 7: $998,000
Prediction 8: $1,998,000
Prediction 9: $875,000
Prediction 10: $3,380,000
Prediction 11: $1,988,000
Prediction 12: $1,199,000


In [32]:
mae = mean_absolute_error(val_y, val_predictions)
print(f"Mean Absolute Error on Validation Set: ${mae:,.0f}")

Mean Absolute Error on Validation Set: $630,833


The Excel file currently has only 45 rows. I need to add a lot more to get a more accurate model. Mean error of $630,833 is awful.

To-do: Add more data.